### DISCUSSION TITLE
How to keep the original type of the parameter when the BaseModel's attribute is a Union?

### DISCUSSION THREAD

**Discussion by arthurazs** [pydantic/pydantic#3970] [Question] [answered]
Mind my example (Python 3.10.4)

```python
>>> import datetime as dt
>>> import pydantic
>>> class DateBug(pydantic.BaseModel):
...     my_date: dt.date | dt.datetime
... 
>>> DateBug(my_date=dt.date(1970, 1, 1))
DateBug(my_date=datetime.date(1970, 1, 1))
>>> DateBug(my_date=dt.datetime(1970, 1, 1, 1))
DateBug(my_date=datetime.date(1970, 1, 1))
```

I want my class to be able to receive both `datetime.date` and `datetime.datetime`, but when I annotate the attribute with `Union[dt.date, dt.datetime]`, the attribute will always be of type `datetime.date`, even if I sent a `datetime.datetime` to it.

How do I maintain the original type of the value in this case?

**Comment by samuelcolvin** [comment, marked as answer]
Swap the order in the union.

(Guess, I'm AFK)

**Reply by arthurazs** [reply]
Simple as that! It worked, thanks

**Reply by samuelcolvin** [reply]
If only all problems could be solved so easily.
